stk/utilities/utilities.py

```diff
@@ -1143,10 +1146,371 @@ def vector_theta(vector1, vector2):
+                return float(nums.search(part2[1].rstrip()).group(0))
+
+    def qonly_dipole_moment(self):
+        """
+        Extracts q only dipole moment in Debye from xTB output.
+
+        Formatting based on the 190418 version of xTB.
+
+        Returns
+        -------
+        :class:`list` of :class:`float`
+            Components of dipole moment as [x, y, z] in a
+            :class:`list` of length 3.
+
+        """
+
+        sample_set = []
+        for i, line in enumerate(self.output_string):
+            if 'molecular dipole:' in line:
+                sample_set = self.output_string[i+2].rstrip()
+
+        # get values from line
+        if 'q only:' in sample_set:
+            return [
+                float(i)
+                for i in sample_set.split(':')[1].split(' ') if i
+            ]
+
+    def full_dipole_moment(self):
+        """
+        Extracts full dipole moment in Debye from xTB output.
+
+        Formatting based on the 190418 version of xTB.
+
+        Returns
+        -------
+        :class:`list` of :class:`float`
+            Components of dipole moment and total magnitude as
+            [x, y, z, total] in a :class:`list` of length 4.
+
+        """
+
+        sample_set = []
+        for i, line in enumerate(self.output_string):
+            if 'molecular dipole:' in line:
+                sample_set = self.output_string[i+3].rstrip()
+
+        # get values from line
+        if 'full:' in sample_set:
+            return [
+                float(i)
+                for i in sample_set.split(':')[1].split(' ') if i
+            ]
+
+    def qonly_quadrupole_moment(self):
+        """
+        Extracts q only traceless quadrupole moment in Debye.
+
+        Formatting based on the 190418 version of xTB.
+
+        Returns
+        -------
+        :class:`list` of :class:`float`
+            Components of quadrupole moment as
+            [xx, xy, xy, xz, yz, zz] in a :class:`list` of length 6.
+
+        """
+
+        sample_set = []
+        for i, line in enumerate(self.output_string):
+            if 'molecular quadrupole (traceless):' in line:
+                sample_set = self.output_string[i+2].rstrip()
+
+        # get values from line
+        if 'q only:' in sample_set:
+            return [
+                float(i)
+                for i in sample_set.split(':')[1].split(' ') if i
+            ]
+
+    def qdip_quadrupole_moment(self):
+        """
+        Extracts q+dip traceless quadrupole moment in Debye.
+
+        Formatting based on the 190418 version of xTB.
+
+        Returns
+        -------
+        :class:`list` of :class:`float`
+            Components of quadrupole moment as
+            [xx, xy, xy, xz, yz, zz] in a :class:`list` of length 6.
+
+        """
+
+        sample_set = []
+        for i, line in enumerate(self.output_string):
+            if 'molecular quadrupole (traceless):' in line:
+                sample_set = self.output_string[i+3].rstrip()
```

Comment:
do we need to keep iterating through these lines after the value has been found? I would assume so since maybe you want the last line, but maybe worth a comment? (applies to other functions too)

Reply:
nah, there should be a break there.